src/Choreography/Internal/Location.hs

```diff
@@ -28,77 +25,71 @@ wrap = Wrap
 
 -- | Unwrap a `Located` value.
 --Unwrapping a empty located value will throw an exception!
-unwrap :: Subset qs ls -> Located ls a -> a
+unwrap :: Subset qs ls -> Located ls a -> a  -- This is wrong! should use membership to avoid empty sets.
 unwrap _ (Wrap a) = a
 unwrap _ Empty    = error "Located: This should never happen for a well-typed choreography."
 
 -- | A unified representation of possibly-distinct homogeneous values owned by many parties.
-newtype Faceted (ls :: [LocTy]) a = Faceted [(LocTm, a)]
+newtype Faceted (ls :: [LocTy]) a = FacetF (forall l. Member l ls -> Located '[l] a)
 
--- | Unwrap a `Faceted` value.
---Unwrapping an empty faceted value will throw an exception!
-mine :: (KnownSymbol l) => Member l ls -> Faceted ls a -> a
-mine l (Faceted facets) = fromMaybe (error "Faceted: This should never happen for a well-typed choreography.")
-                                    $ toLocTm l `lookup` facets
-
-
--- GDP has its own list logic, but IDK how to use it...
-data IsMember (x :: k) (xs :: [k]) where {}
-type Member x xs = Proof (IsMember x xs)
-data IsSubset (xs :: [k]) (ys :: [k]) where {}
-type Subset xs ys = Proof (IsSubset xs ys)
-instance Reflexive IsSubset where {}
-instance Transitive IsSubset where {}
-
-class ExplicitMember (x :: k) (xs :: [k]) where
-  explicitMember :: Member x xs
-instance {-# OVERLAPPABLE #-} (ExplicitMember x xs) =>  ExplicitMember x (y ': xs) where
-  explicitMember = inSuper consSet explicitMember
-instance {-# OVERLAPS #-} ExplicitMember x (x ': xs) where
-  explicitMember = axiom
+unsafeFacet :: [Maybe a] -> Member l ls -> Located '[l] a
+unsafeFacet (Just a : _) First = Wrap a
+unsafeFacet (Nothing : _) First = Empty
+unsafeFacet (_ : as) (Later l) = unsafeFacet as l
+unsafeFacet [] _ = error "The provided list isn't long enough to use as a Faceted over the intended parties."
+
+
+data Member (x :: k) (xs :: [k]) where
```

Comment:
This is nice! What do you think about some documentation or examples? It took me a bit to understand this after being away from this project for a while:

```suggestion
{- | A proof that a location is a member of a list of locations.
 __Examples:__

@
aliceIsFirst = First :: Member "alice" '["alice", "bob"]
bobIsSecond = Later First ::  Member "bob" '[ "alice", "bob"]
@
-}
data Member (x :: k) (xs :: [k]) where
```

Reply:
I can do the others if you think this would be useful

Reply:
Documentation is obviously lacking and good. I think the example you have above is over-verbose; might be better to just say "The structure of a `Member l ls` encodes a natural/Peano number, but the signatures of the constructors enforce that said number is an index into `ls` at which `l` appears".